core/server/models/post.js

```diff
@@ -139,9 +139,10 @@ Post = ghostBookshelf.Model.extend({
             }
         }
 
-        if (this.hasChanged('slug') || !this.get('slug')) {
+        // If the status is not published and has never been published (published at never been set), update slug with title
+        if (this.get('status') === 'draft' && !this.get('published_at')) {
```

Comment:
Could use a little help here. Functionality works, but the `Post API Edit can edit a new published post and unpublish` test always fails. I can get the test to work if i change this line to be:

    if ((this.get('status') === 'draft' && !this.get('published_at')) || (this.hasChanged('slug') || !this.get('slug')))

The test passes with that, but that is not needed for the functionality, so I didn't want to add it in just to have the test pass if it was unnecessary. Does anyone have an idea on why the test isn't passing and either how I can fix it or if i should just add that extra logic into the if statement?